src/main/java/com/google/sps/servlets/FreeTimeServlet.java

```diff
@@ -0,0 +1,209 @@
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.model.GmailClientImpl;
+import com.google.sps.utility.DateInterval;
+import com.google.sps.utility.FreeTimeUtility;
+import com.google.sps.utility.JsonUtility;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.StringTokenizer;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.mail.MessagingException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** GET function responds JSON string containing potential events to read mail. */
+@WebServlet("/plan-mail")
+public class FreeTimeServlet extends AuthenticatedHttpServlet {
+  private final CalendarClientFactory calendarClientFactory;
+  private final GmailClientFactory gmailClientFactory;
+
+  /** Create servlet with default CalendarClient and Authentication Verifier implementations */
+  public FreeTimeServlet() {
+    calendarClientFactory = new CalendarClientImpl.Factory();
+    gmailClientFactory = new GmailClientImpl.Factory();
+  }
+
+  /**
+   * Create servlet with explicit implementations of CalendarClient and AuthenticationVerifier
+   *
+   * @param authenticationVerifier implementation of AuthenticationVerifier
+   * @param calendarClientFactory implementation of CalendarClientFactory
+   */
+  public FreeTimeServlet(
+      AuthenticationVerifier authenticationVerifier,
+      CalendarClientFactory calendarClientFactory,
+      GmailClientFactory gmailClientFactory) {
+    super(authenticationVerifier);
+    this.calendarClientFactory = calendarClientFactory;
+    this.gmailClientFactory = gmailClientFactory;
+  }
+
+  /**
+   * Returns string containing the some of the user's free time intervals, where events can be
+   * created
+   *
+   * @param request Http request from the client. Should contain idToken and accessToken
+   * @param response 403 if user is not authenticated, or Json string with the user's events
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    CalendarClient calendarClient = calendarClientFactory.getCalendarClient(googleCredential);
+    long fiveDaysInMillis = TimeUnit.DAYS.toMillis(5);
+    Date timeMin = calendarClient.getCurrentTime();
+    Date timeMax = Date.from(timeMin.toInstant().plus(Duration.ofDays(5)));
+    List<Event> calendarEvents = getEvents(calendarClient, timeMin, timeMax);
+
+    int personalBeginHour = 7;
+    int workBeginHour = 10;
+    int workEndHour = 18;
+    int personalEndHour = 23;
+    int numDays = 5;
```

Comment:
Maybe a comment to explain these numbers? I can predict what `numDays` means but not so much the others.

Reply:
`Resolved`

Reply:
Thanks for adding the documentation! I think these should be class constants for now - you can always change that if/when you supply some of the values in the request.